wandb/apis/normalize.py

```diff
@@ -24,28 +23,6 @@ def wrapper(*args, **kwargs):
             return func(*args, **kwargs)
         except requests.HTTPError as err:
             raise CommError(err.response, err)
-        except RetryError as err:
```

Comment:
Was this path not used? if it was, what is the replacement?

Reply:
Looking closer at this, it looks like the old behavior did a blanket exception catch so if requests happens to throw an exception it would get rethrown here,  without this, there is a risk of an unhandled exception (though the only I could find look unlikely in our cases (too many redirects, etc)

Reply:
This path was used for generating better graphql errors when the server returned an "error" payload.  Now the GQLClient just raises a generic `Exception` with the first error in the payload as the message.  We may want to raise a special error class for that case, I'll chew on it a bit.

Reply:
If you want to merge it now and let things shake out, that might be good too, as long as we closely think about the implications in this path as it is the highest risk of this change.

Reply:
@raubitsj I did some more work here.  Specifically [this commit](https://github.com/wandb/client/pull/2419/commits/152f887293a2aa02f0023cd2b92133c255c08ffc) would be good to review.  It loosens the graphql-core requirement so we could release this on the 0.11.x branch if we want.

`RetryError` used to be raised by gql when it's own internal retry logic failed.  I've now made `execute` only raise `requests.RequestExeception` errors.  I also handled two cases that used to suck.

1. If we get error details from the server, the `requests.HTTPError` error that we raise will have a better message.  This is especially useful in the public api.  For example:

## Before

```
runs = api.runs("foo/bar", filters={"summary.foo": "bar"})
HTTPError: 400 Client Error: Bad Request for url: https://api.wandb.ai/graphql
```

## After

```
runs = api.runs("foo/bar", filters={"summary.foo": "bar"})
HTTPError: Error 1054: Unknown column 'summary' in 'where clause'
```

2. If the server doesn't return valid JSON we raise a `requests.HTTPError: 500` with a message "http://localhost:8080/graphql did not return valid JSON" to handle invalid proxies or shutdown local instances.